File: livingwell-frontend/src/types/ai.ts
export interface PropertyDefaultsRequest {
  address: string;
  zoning: string;
  city?: string;
}

export interface PropertyDefaultsResponse {
  estimated_lot_size: number;
  max_buildable_area: number;
  recommended_units: number;
  estimated_cost_per_sqft: number;
  reasoning: string;
}

export interface RiskItem {
  category: string;
  severity: 'Low' | 'Medium' | 'High' | 'Critical' | string;
  description: string;
  mitigation: string;
}

export interface RiskAnalysisResponse {
  overall_risk_score: number;
  summary: string;
  risks: RiskItem[];
}

File: backend/app/schemas/ai.py
"""Pydantic schemas for AI endpoints."""
from pydantic import BaseModel
from typing import Optional


class PropertyDefaultsRequest(BaseModel):
    address: str
    zoning: str
    city: str = "Calgary"


class PropertyDefaultsResponse(BaseModel):
    estimated_lot_size: float
    max_buildable_area: float
    recommended_units: int
    estimated_cost_per_sqft: float
    reasoning: str


class RiskAnalysisRequest(BaseModel):
    property_id: int


class RiskItemSchema(BaseModel):
    category: str
    severity: str
    description: str
    mitigation: str


class RiskAnalysisResponse(BaseModel):
    overall_risk_score: int
    summary: str
    risks: list[RiskItemSchema]
